src/Qir/Runtime/lib/QIR/context.cpp

```diff
@@ -46,21 +31,92 @@ namespace Quantum
             g_context->allocationsTracker->CheckForLeaks();
         }
 
-        ResultOne = nullptr;
-        ResultZero = nullptr;
         g_context.reset(nullptr);
     }
 
-    QirExecutionContext::QirExecutionContext(IRuntimeDriver* sim, bool trackAllocatedObjects)
-        : driver(sim)
+    QirExecutionContext::QirExecutionContext(IRuntimeDriver* simulator, bool trackAllocatedObjects)

```

Comment:
nit: This shouldn't really be named simulator, as it doesn't have to be one. It could be a driver that sends instructions to hardware or collections gates to output to something else. Not something I think you need to change here, but it makes it seem like this runtime infrastructure is specific to simulation, which it is not.

Reply:
Renamed.